web/auth.go

```diff
@@ -413,79 +423,120 @@ type tokenResponse struct {
-	defer cancel()
-
-	_, span := trace.StartSpan(ctx, r.URL.Path)
-	defer span.End()
-
-	token, err := r.Cookie("viam.auth.token")
-	if HandleError(w, err, h.logger, "getting token cookie") {
-		return
+func getBearerToken(req *http.Request) string {
+	authHeader := req.Header.Get("Authorization")
+	if authHeader == "" {
+		return ""
 	}
 
-	refresh, err := r.Cookie("viam.auth.refresh")
-	if HandleError(w, err, h.logger, "getting refresh cookie") {
-		return
+	parts := strings.Split(authHeader, " ")
+	if len(parts) == 2 && parts[0] == "Bearer" {
+		return parts[1]
 	}
 
-	expiry, err := r.Cookie("viam.auth.expiry")
-	if HandleError(w, err, h.logger, "getting expiry cookie") {
-		return
-	}
+	return ""
+}
 
-	response := &tokenResponse{
-		AccessToken:  token.Value,
-		RefreshToken: refresh.Value,
-		Expiry:       expiry.Value,
+// getAuthCookieValues reads the authentication cookie values as a /token response
+// before clearing the cookies.
+func getAndClearAuthCookieValues(w http.ResponseWriter, r *http.Request) *tokenResponse {
+	token, err := r.Cookie(ViamTokenCookie)
+	if err != nil || token.Value == "" {
+		return nil
 	}
 
-	w.Header().Set("Content-Type", "application/json")
-	data, err := json.Marshal(response)
+	refresh, err := r.Cookie(ViamRefreshCookie)
+	// TODO: Check if refresh is empty when implemented, always empty now
 	if err != nil {
-		temp := errors.New("failed to verify marshal token data: " + err.Error())
-		w.WriteHeader(http.StatusInternalServerError)
-		_, err = w.Write([]byte(temp.Error()))
-		if err != nil {
-			utils.UncheckedError(err)
-		}
-		h.logger.Error(temp)
-		return
+		return nil
+	}
+
+	expiry, err := r.Cookie(ViamExpiryCookie)
+	if err != nil || expiry.Value == "" {
+		return nil
 	}
 
 	http.SetCookie(w, &http.Cookie{
-		Name:     "viam.auth.token",
+		Name:     ViamTokenCookie,
 		Value:    "",
 		Path:     "/",
 		MaxAge:   -1,
-		Secure:   r.TLS != nil,
 		SameSite: http.SameSiteLaxMode,
 		HttpOnly: true,
 	})
 
 	http.SetCookie(w, &http.Cookie{
-		Name:     "viam.auth.refresh",
+		Name:     ViamRefreshCookie,
 		Value:    "",
 		Path:     "/",
 		MaxAge:   -1,
-		Secure:   r.TLS != nil,
 		SameSite: http.SameSiteLaxMode,
 		HttpOnly: true,
 	})
 
 	http.SetCookie(w, &http.Cookie{
-		Name:     "viam.auth.expiry",
+		Name:     ViamExpiryCookie,
 		Value:    "",
 		Path:     "/",
 		MaxAge:   -1,
-		Secure:   r.TLS != nil,
 		SameSite: http.SameSiteLaxMode,
 		HttpOnly: true,
 	})
 
-	_, err = w.Write(data)
-	utils.UncheckedError(err)
+	return &tokenResponse{
+		AccessToken:  token.Value,
+		RefreshToken: refresh.Value,
+		Expiry:       expiry.Value,
+	}
+}
+
+func (h *tokenHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
+	defer cancel()
+
+	_, span := trace.StartSpan(ctx, r.URL.Path)
+	defer span.End()
+
+	current := getBearerToken(r)
+	data := getAndClearAuthCookieValues(w, r)
```

Comment:
note that theres a small change in behavior here where we will still clear cookies even if the json marshling fails. i think this is fine but calling it out

Reply:
Good callout. I agree it's fine, if the marshalling fails it probably means something was broken/corrupted in the auth flow and it is probably best to start again with a clean state.